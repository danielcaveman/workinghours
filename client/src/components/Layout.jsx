import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Header from "../components/Header";
import Menu from "../components/Menu";
import Main from "../components/Main";
import GlobalStyles from "../styles/global";

const StyledContainer = styled.div`
  display: grid;
  grid-template-areas:
    "menu header header header header header"
    "menu main main main main main";
  grid-template-rows: 7rem auto;
  grid-template-columns: 30rem auto;
  width: 100vw;
  height: 100vh;
`;

function MainPage({ children }) {
  return (
    <>
      <GlobalStyles />
      <StyledContainer>
        <Header />
        <Menu />
        <Main child={children} />
      </StyledContainer>
    </>
  );
}

MainPage.propTypes = {
  children: PropTypes.array,
};

export default MainPage;
