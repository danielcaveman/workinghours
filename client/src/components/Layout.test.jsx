import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Layout from "./Layout";

describe("<Layout />", () => {
  it("Should render layout component correctly", () => {
    render(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    );

    const header = screen.getByRole("banner");
    expect(header).toBeInTheDocument();

    const aside = screen.getByRole("complementary");
    expect(aside).toBeInTheDocument();

    const nav = screen.getByRole("navigation");
    expect(nav).toBeInTheDocument();

    const main = screen.getByRole("main");
    expect(main).toBeInTheDocument();
  });
});
