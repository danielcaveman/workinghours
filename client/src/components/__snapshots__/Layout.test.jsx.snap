// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Should match snapshot 1`] = `
<div>
  <div
    class="sc-fubCfw ihTred"
  >
    <header
      class="sc-bdfBwQ QtUwF"
    />
    <aside
      class="sc-gsTCUz gvzgmn"
    >
      <div
        class="sc-dlfnbm lgzpeJ"
      />
      <nav
        class="sc-hKgILt bVFLLn"
      >
        <ul
          class="sc-eCssSg bdhGbw"
        >
          <li
            class="sc-jSgupP"
          >
            <a
              aria-current="page"
              class="sc-gKsewC kJkXnr active"
              href="/"
            >
              Main Page
            </a>
          </li>
        </ul>
      </nav>
    </aside>
    <main
      class="sc-iBPRYJ gRKJzj"
    />
  </div>
</div>
`;
